test/correctness/simd_op_check.cpp

```diff
@@ -1871,6 +1871,16 @@ struct Test {
         check("vmpa(v*.h,r*.b)", hvx_width/2, 2*i32(i16_1) + 3*i32(i16_2));
         check("v*.w += vmpa(v*.h,r*.b)", hvx_width/2, 2*i32(i16_1) + 3*i32(i16_2) + i32_1);
 
+        check("v*:*.h = vtmpy(v*:*.ub, r*.b)", hvx_width/1, 2*i16(in_u8(x + hvx_width - 1)) + 3*i16(in_u8(x + hvx_width)) + i16(in_u8(x + hvx_width + 1)));
```

Comment:
@dsharletg - If he chooses a positive value for PADD such as 128, then the domain of 'f' is always inside the image because x will be (128, W-128). I don't think then we need to pad the buffers in apps/simd_op_check/drivers.cpp, do we?
@aankit-ca  - can you confirm this by running apps/simd_op_check?